packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -1431,6 +1432,59 @@ describe('Table row selection', () => {
                     );
                 });
 
+                describe('with SHIFT pressed in the document then let go outside the window', () => {
+                    beforeEach(() => {
+                        const shiftKeyDownEvent = new KeyboardEvent('keydown', {
+                            key: keyTab, // could be any key
+                            shiftKey: true
+                        } as KeyboardEventInit);
+                        document.dispatchEvent(shiftKeyDownEvent);
+                        window.dispatchEvent(new FocusEvent('blur'));
+                        // No SHIFT keyup event. This simulates the user letting go of the SHIFT key outside the window.
+                    });
+
+                    it('selects only the rows whose checkboxes were clicked', async () => {
+                        pageObject.clickRowSelectionCheckbox(0);
+                        pageObject.clickRowSelectionCheckbox(3);
+                        await waitForUpdatesAsync();
+
+                        const selectedRecordIds = await element.getSelectedRecordIds();
+                        expect(selectedRecordIds).toEqual(
+                            jasmine.arrayWithExactContents([
+                                simpleTableData[0].id,
+                                simpleTableData[3].id
+                            ])
+                        );
+                    });
+
+                    it('selects only the rows that SPACE was pressed on', async () => {
+                        const downArrowKeyEvent = new KeyboardEvent('keydown', {
```

Comment:
nit: wanna use [the `sendKeyDownEvent` test utility](https://github.com/ni/nimble/blob/f47da47d1ee6104e9e4c913c8e1fbca321825e9d/packages/nimble-components/src/utilities/tests/component.ts#L8) in these new tests instead of constructing and dispatching event objects?

There are plenty of existing places we don't do this, but it is nice to centralize it and abstract away the details if possible.

Reply:
I wasn't excited about adding a bunch of extra `waitForUpdatesAsync` calls, so I made a `sendKeyDownEvents` variant to send a sequence of keystrokes then wait once.